File: multi/1_Two_sum/Two_Sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    done = {}
    nums.each_with_index do |element, i|
        if done.has_key?(target-nums[i])
            return [done[target-nums[i]], i]
        end
        done[nums[i]] = i
    end
end

File: multi/1_Two_sum/Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var done = {};
    var sums = [];
    for(var i=0; i<nums.length; i++){
        if((target-nums[i]) in done){
            sums.push(done[target-nums[i]]);
            sums.push(i);
            return sums;
        }
        done[nums[i]] = i;
    };

    return sums;
};